contracts/src/v0.8/dev/vrf/VRFV2PlusWrapper.sol

```diff
@@ -358,10 +358,12 @@ contract VRFV2PlusWrapper is ConfirmedOwner, TypeAndVersionInterface, VRFConsume
   function onTokenTransfer(address _sender, uint256 _amount, bytes calldata _data) external onlyConfiguredNotDisabled {
     require(msg.sender == address(s_link), "only callable from LINK");
 
-    (uint32 callbackGasLimit, uint16 requestConfirmations, uint32 numWords) = abi.decode(
+    (uint32 callbackGasLimit, uint16 requestConfirmations, uint32 numWords, bytes memory extraArgs) = abi.decode(
       _data,
-      (uint32, uint16, uint32)
+      (uint32, uint16, uint32, bytes)
     );
+    bool nativePayment = abi.decode(removeDomainSeparator(extraArgs), (bool));
+    require(!nativePayment, "nativePayment not false in onTokenTransfer call");
```

Comment:
I think you need to have a `require` statement prior to this that checks the length of the array perhaps? Then you can do a single element access instead to check that the appropriate bit is set:

```suggestion
	if (extraArgs.length < 36) {
		revert IncorrectExtraArgsLength(EXPECTED_LENGTH, extraArgs.length);
	}
    bool nativePayment = extraArgs[35] == 1
    if (nativePayment) {
    	revert NativePaymentInOnTokenTransfer();
    }
```

Reply:
Fixed this as per suggestion